cartridge/test-helpers/etcd.lua

```diff
@@ -0,0 +1,114 @@
+--- Class to run and manage etcd node.
+--
+-- @classmod cartridge.test-helpers.etcd
+
+local checks = require('checks')
+local log = require('log')
+local fun = require('fun')
+local httpc = require('http.client')
+
+local luatest = require('luatest')
+local Process = require('luatest.process')
+
+-- Defaults.
+local Etcd = {
+    workdir = nil,
+    etcd_path = nil,
+    name = 'default',
+    peer_url = 'http://127.0.0.1:17001',
+    client_url = 'http://127.0.0.1:14001',
+    args = {},
+    env = {},
+}
+
+function Etcd:inherit(object)
+    setmetatable(object, self)
+    self.__index = self
+    return object
+end
+
+--- Build etcd node object.
+-- @param object
+-- @string object.name Human-readable node name.
+-- @string object.workdir Path to the data directory.
+-- @string object.etcd_path Path to the etcd executable.
+-- @string object.peer_url URL to listen on for peer traffic.
+-- @string object.client_url URL to listen on for client traffic.
+-- @tab[opt] object.env Environment variables passed to the process.
+-- @tab[opt] object.args Command-line arguments passed to the process.
+-- @return object
+function Etcd:new(object)
+    checks('table', {
+        name = '?string',
+        workdir = 'string',
+        etcd_path = 'string',
+        peer_url = '?string',
+        client_url = '?string',
+        args = '?table',
+        env = '?table',
+    })
+    self:inherit(object)
+    object:initialize()
+    return object
+end
+
+function Etcd:initialize()
+    self.env = fun.chain({
+        ETCD_NAME = self.name,
+        ETCD_DATA_DIR = self.workdir,
+        ETCD_LISTEN_PEER_URLS = self.peer_url,
+        ETCD_LISTEN_CLIENT_URLS = self.client_url,
+        ETCD_ADVERTISE_CLIENT_URLS = self.client_url,
+    }, self.env):tomap()
+end
+
+--- Start the node.
+function Etcd:start()
+    local env = table.copy(os.environ())
```

Comment:
```
tarantool> os.environ()['a'] = 2
---
...

tarantool> os.environ()['a']
---
- null
...
```

Seems table.copy is redundant

Reply:
You're right, it [returns](https://github.com/tarantool/tarantool/blob/0ab21ac9eeaaae2aa0aef5e598d374669f96df9e/src/lua/env.lua#L12-L28) new table every time. Fixed